master/src/main/proto/Resource.proto

```diff
@@ -83,6 +83,7 @@ message DiskInfo {
   required int32 status = 5;
   required int64 avgFetchTime = 6;
   required int32 storageType =7;
+  required int64 totalSpace = 8;
```

Comment:
Using optional will be better. Adding a required field will block the rolling upgrade.